src/Illuminate/Database/Eloquent/Builder.php

```diff
@@ -1178,15 +1178,16 @@ protected function addUpdatedAtToUpsertColumns(array $update)
     /**
      * Delete records from the database.
      *
-     * @return mixed
+     * @param  mixed  $id
+     * @return mixed|int
      */
-    public function delete()
+    public function delete($id = null)
```

Comment:
This is a breaking change on the method signature unfortunately.

Reply:
@driesvints  how is this breaking? in case someone overrides the builder class?
But the parent class already had this method?

Reply:
Yes, if someone overwrite the builder class' `delete` method their code will break.